sandbox/apparmor/apparmor.go

```diff
@@ -51,6 +53,30 @@ const (
 	Full
 )
 
+func setupConfCacheDirs(newrootdir string) {
+	ConfDir = filepath.Join(newrootdir, "/etc/apparmor.d")
+	CacheDir = filepath.Join(newrootdir, "/var/cache/apparmor")
+
+	SystemCacheDir = filepath.Join(ConfDir, "cache")
+	exists, isDir, _ := osutil.DirExists(SystemCacheDir)
+	if !exists || !isDir {
+		// some systems use a single cache dir instead of splitting
+		// out the system cache
+		SystemCacheDir = CacheDir
+	}
+}
+
+func init() {
+	dirs.AddRootDirCallback(setupConfCacheDirs)
+	setupConfCacheDirs(dirs.GlobalRootDir)
```

Comment:
It would be nice if adding a callback automatically invoked it once.

Reply:
we would need to rename AddRootDirCallback for clarity and I don't have a suggestion atm that isn't a mouthful, please leave this alone for now.